onnxscript/type_annotation.py

```diff
@@ -82,6 +82,25 @@ def pytype_to_attrtype(
     return None
 
 
+def base_type_is_bool(pytype: TypeAnnotationValue) -> bool:
+    """Returns True if base type of pytype is bool, False otherwise."""
+    pytype = _remove_annotation(pytype)
+    if pytype in _PYTYPE_TO_ATTRTYPE_MAP:
+        return pytype is bool
+    type_constructor = typing.get_origin(pytype)
+    # Remove Optional wrapper if present, which is represented as an Union[..., type(None)]
```

Comment:
Potentially use https://github.com/microsoft/onnxscript/blob/a53c5aeece3aca85964477a6bec6facc0a1f5644/onnxscript/type_annotation.py#L160-L168 for optional checks? Looks like `typing.get_origin(pytype) is Optional` from 3.10